test/python/algorithms/eigensolvers/test_vqd.py

```diff
@@ -230,6 +227,9 @@ def run_check():
         with self.subTest("Optimizer replace"):
             vqd.optimizer = L_BFGS_B()
             run_check()
+        with self.subTest("SPSA replace"):
+            vqd.optimizer = SPSA()
```

Comment:
It might be nice so manually set the batch size here so we are sure the breaking case is covered, even if the default in SPSA changes later 🙂